pandas/core/generic.py

```diff
@@ -4304,8 +4304,9 @@ def fillna(self, value=None, method=None, axis=None, inplace=False,
                 elif not is_list_like(value):
                     pass
                 else:
-                    raise ValueError("invalid fill value with a %s" %
-                                     type(value))
+                    raise TypeError('"value" parameter must be a scalar, dict '
+                                    'or Series, but you passed a '
```

Comment:
ideally we would have a parametriezed test that hits this (with multiple invalid things that should raise)

Reply:
Yeah, as I said above, I have included tests in ``test_fillna_categorical_raise()`` in ``tests/series/test_missing.py`` to check for the TypeError when the user passes a list, tuple, or dict fill ``value``. But the tests aren't parametrized.